core/state/statedb.go

```diff
@@ -866,27 +905,58 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 	s.clearJournalAndRefund()
 }
 
+func (s *StateDB) collectReadStorageAccessLists() {
```

Comment:
This will only work if snapshots are enabled, no? Otherwise the readPrefetcher is a noop and reads are accumulated through the actual state object trie. So we'd need to pull the access list from that?

Reply:
Annoyingly, it could even happen in theory that some items are looked up from the state snapshot and some from the trie (if snapshot lookup fails), so we always need to merge the two.